### PR TITLE
Adds QUnit tests to Jetpack

### PR DESCRIPTION
**Pull request by zinigor**
This PR mostly copies what WP Core does to test JavaScript with some simplifications. There is one new unit tests that is only there to show how (and if) JS tests would work in Jetpack.

### CODE REVIEW

File: tests/qunit/index.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	<head>
+		<title>Jetpack QUnit Test Suite</title>
+
+		<!-- Dependencies -->
+		<script src="../../../../../../src/wp-includes/js/jquery/jquery.js"></script>
+		<script src="../../../../../../src/wp-includes/js/jquery/ui/core.js"></script>
+		<script src="../../../../../../src/wp-includes/js/underscore.min.js"></script>
+		<script src="../../../../../../src/wp-includes/js/backbone.min.js"></script>
+		<script src="../../../../../../src/wp-includes/js/wp-backbone.js"></script>
+		<script src="../../../../../../src/wp-includes/js/zxcvbn.min.js"></script>
```
Comment (eliorivero) [inline tests/qunit/index.html]: @zinigor perhaps we can remove zxcvbn.min.js since we don't use it in Jetpack.

**Comment by zinigor** [comment]
Good call, @eliorivero, thanks! I have removed the library.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "Jetpack",
-  "version": "3.9.0-alpha",
+  "version": "3.8.2-beta",
```
Comment (samhotchkiss) [inline package.json]: This probably shouldn't change, eh?

**Comment by zinigor** [comment]
@samhotchkiss thanks for the catch! I fixed it, as well as updated the vendor files by deleting them and using code from `npm` packages, so we don't introduce a lot of foreign code into the repo.

**Pull request merged**